### PR TITLE
Open up curations, with a hash

### PR DESCRIPTION
**Pull request by pagreene**
This PR makes it so that users with any credentials (an API key or account) can get all the curations, however without extra permissions the identities will be hashed and salted.

This PR adds a new config environment variable: `INDRA_DB_API_CURATOR_SALT`.

This PR also switches to using the `pathlib.Path` object for file paths.

### CODE REVIEW

File: indra_db_service/api.py [hunk 1]
```diff
@@ -193,42 +194,33 @@ def search():
         vue_style=vue_style, stmt_types_json=stmt_types_json)
 
 
+suf_ct_map = {'.js': 'application/javascript', '.css': 'text/css'}
+
+
 @app.route('/data-vis/<path:file_path>')
 def serve_data_vis(file_path):
-    full_path = path.join(HERE, 'data-vis/dist', file_path)
+    full_path = HERE / 'data-vis' / 'dist' / file_path
```
Comment (cthoyt) [inline indra_db_service/api.py:202]: This might be the most legible:

```suggestion
    full_path = HERE.joinpath('data-vis', 'dist', file_path)
```

Comment (pagreene) [reply indra_db_service/api.py:202]: Personally, I disagree. I think "/" is well established as the divider between path segments, making the statement unambiguous. And someone looking at it that doesn't "know" programming would in fact be less confused and have an easier time interpreting it, which to my mind makes it more "pythonic".

That said, it is pretty subjective, and I am most certainly open to rebuttals :)

Comment (cthoyt) [reply indra_db_service/api.py:202]: I don't have a strong opinion. I think there's no mistaking what's going on with the slashes

**Comment by cthoyt** [comment]
Maybe would be good to add a code example in here somewhere on how to take advantage of the curation endpoint in combination with the INDRA `filter_by_curation()` function https://github.com/sorgerlab/indra/blob/bc39dae6849b1fd484d83eabb3d2afee963a6298/indra/tools/assemble_corpus.py#L1669

**Comment by kkaris** [comment]
Just tested locally, will test on db.indra.bio/dev

**Pull request merged**